_docs/migration_guides/upgrading_to_terragrunt_0.12.x.md

```diff
@@ -0,0 +1,174 @@
+# Upgrading to Terragrunt 0.12.x
```

Comment:
Great idea to break this out into a separate file. Perhaps we should do that in follow-up PRs for more of the README? That might keep the file size down and remove the need for so many TOCs...